Analysis/Tasks/SkimmingTutorials/spectraTPCProvider.cxx

```diff
@@ -30,30 +31,131 @@ struct TPCSpectraProviderTask {
 
   //Produces<aod::LFCollisions> outputCollisions; //currently it seems in the spectraTPC task no loop over the collision is made. Leave this here in case it will be added
   Produces<aod::LFTracks> outputTracks;
+  Produces<aod::LFSmallTracks> outputSmallTracks;
+  Produces<aod::LFSingleTracks> outputSingleTracks;
 
   Configurable<float> vertexZCut{"vertexZCut", 10.0f, "Accepted z-vertex range"};
   Configurable<float> trackEtaCut{"trackEtaCut", 0.8f, "Eta range for tracks"};
-  Configurable<float> trackPtCut{"trackPtCut", 0.0f, "Pt range for tracks"};
+  Configurable<float> trackPtCut{"trackPtCut", 0.1f, "Pt range for tracks"};
+
+  Configurable<bool> saveTracks{"saveTracks", false, "Save large LF tracks table"};
+  Configurable<bool> saveSmallTracks{"saveSmallTracks", false, "Save small LF tracks table"};
+  Configurable<bool> saveSingleTracks{"saveSingleTracks", false, "Save single species LF tracks table"};
+  Configurable<int> species1{"species1", 0, "First particle species to be kept"};
+  Configurable<bool> species1NsigmaSelection{"species1NsigmaSelection", false, "select on species 1 Nsigma"};
+  Configurable<float> species1NsigmaLow{"species1NsigmaLow", -3., "species 1 Nsigma Lower Bound"};
+  Configurable<float> species1NsigmaHigh{"species1NsigmaHigh", 3., "species 1 Nsigma Higher Bound"};
+  Configurable<int> species2{"species2", 0, "Second particle species to be kept"};
+  Configurable<bool> species2NsigmaSelection{"species2NsigmaSelection", false, "select on species 2 Nsigma"};
+  Configurable<float> species2NsigmaLow{"species2NsigmaLow", -3., "species 2 Nsigma Lower Bound"};
+  Configurable<float> species2NsigmaHigh{"species2NsigmaHigh", 3., "species 2 Nsigma Higher Bound"};
+  Configurable<int> species3{"species3", 0, "Third particle species to be kept"};
+  Configurable<bool> species3NsigmaSelection{"species3NsigmaSelection", false, "select on species 3 Nsigma"};
+  Configurable<float> species3NsigmaLow{"species3NsigmaLow", -3., "species 3 Nsigma Lower Bound"};
+  Configurable<float> species3NsigmaHigh{"species3NsigmaHigh", 3., "species 3 Nsigma Higher Bound"};
+  Configurable<int> species4{"species4", 0, "Fourth particle species to be kept"};
+  Configurable<bool> species4NsigmaSelection{"species4NsigmaSelection", false, "select on species 4 Nsigma"};
+  Configurable<float> species4NsigmaLow{"species4NsigmaLow", -3., "species 4 Nsigma Lower Bound"};
+  Configurable<float> species4NsigmaHigh{"species4NsigmaHigh", 3., "species 4 Nsigma Higher Bound"};
+
+  template <typename T>
+  float SpeciesNSigma(const T& track, int specie)
+  {
+    uint32_t pNsigma = 0xFFFFFF00; //15 bit precision for Nsigma
+    if (specie == 0) {
```

Comment:
This will be evaluated on each track and is slow. Ideally one templates the specie parameter and makes the if "constexpr" then it is done at compile time. Nicolo put a similar solution for the tinyPID task.

Reply:
But then some of your code below doesn't work... (where one selects which particle to save)

Reply:
is reading the 9 sigma values for each track much faster than going through the if statements? If so for each track I can fill an array with all the Nsigma values and then just select the index in the array corresponding to specie

Reply:
No, it is not faster. We can keep it this way.